core/src/main/java/org/apache/calcite/rel/rules/HyperGraph.java

```diff
@@ -369,25 +388,58 @@ private static boolean isOverlapEdge(HyperEdge edge, long subset) {
     return joinType;
   }
 
-  public RexNode extractJoinCond(RelNode left, RelNode right, List<HyperEdge> edges) {
+  /**
+   * Restore join condition from hyper edges.
```

Comment:
I think that a better description is "build an RexNode expression for the predicate corresponding to a set of hyperedges".